src/plugins/data_source_management/public/components/edit_data_source/components/edit_form/edit_data_source_form.tsx

```diff
@@ -118,16 +118,16 @@ export class EditDataSourceForm extends React.Component<EditDataSourceProps, Edi
       this.setState({
         title,
         description: description || '',
-        endpoint,
+        endpoint: endpoint || '',
         auth: {
           type: auth.type,
           credentials: {
             username: authTypeCheckResults.isUserNamePassword ? auth.credentials?.username : '',
             password: authTypeCheckResults.isUserNamePassword ? this.maskedPassword : '',
             service: authTypeCheckResults.isSigV4
-              ? auth.credentials?.service || SigV4ServiceName.OpenSearch
-              : '',
-            region: authTypeCheckResults.isSigV4 ? auth.credentials!.region : '',
+              ? (auth.credentials?.service as SigV4ServiceName) || SigV4ServiceName.OpenSearch
+              : ('' as SigV4ServiceName),
+            region: authTypeCheckResults.isSigV4 ? String(auth.credentials?.region) || '' : '',
```

Comment:
This expression `String(auth.credentials?.region) || ''` evaluates to `"undefined"` string when `auth.credentials?.region = undefined` which may not quite as you expected. I guess what you intend to do was:
```typescript
String(auth.credentials?.region || '')
```

Reply:
fixed, thanks!